### PR TITLE
Sets recommended max Xmx 32G when set_recommended_max_xmx experiment is enabled

### PR DESCRIPTION
**Pull request by arvindram03**
We set the Xmx to the recommended max value(32G) with set_recommended_max_xmx is enabled. Under 32G JVM enables CompressedOops. CompressedOops utilizes memory more efficiently, and has positive impact on GC performance and cache hit rate. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/java/container/boot.go [hunk 1]
```diff
@@ -266,9 +267,12 @@ func makePipelineOptionsFile(options string) error {
...
 // that value, it returns 1GB. This is an imperfect heuristic. It aims to
 // ensure there is memory for non-heap use and other overhead, while also not
-// underutilizing the machine.
-func heapSizeLimit(info *fnpb.ProvisionInfo) uint64 {
-	if size, err := syscallx.PhysicalMemorySize(); err == nil {
+// underutilizing the machine. if set_recommended_max_xmx experiment is enabled, sets xmx 
+// to 32G
+func heapSizeLimit(info *fnpb.ProvisionInfo, setRecommendedMaxXmx bool) uint64 {
+	if setRecommendedMaxXmx {
+		return 32 << 30
```
Comment (arunpandianp) [inline sdks/java/container/boot.go:274]: It'll be flexible if we allow passing the xmx value via the experiment.
eg: `--experiments= set_recommended_xmx=<xmx bytes>`

Comment (arvindram03) [reply sdks/java/container/boot.go:274]: We do not want to expose the plug too much with this change as well. Which later becomes hard to control. In future, might be a generic way to control all the jvm args which will be covering this as well.

Comment (Abacn) [reply sdks/java/container/boot.go:274]: could you please include the description regarding why 32G as a comment here?

Comment (Abacn) [reply sdks/java/container/boot.go:274]: thanks, I mean code comment

Comment (arvindram03) [reply sdks/java/container/boot.go:276]: Done

**Comment by arvindram03** [comment]
assign set of reviewers

**Review by Abacn — APPROVED** [review]

**Comment by arvindram03** [comment]
Thanks for the swift approval.

**Pull request merged**